File: week2_algorithmic_warmup/lcm.go
package main

import "fmt"

func gcd(s int, l int) int {
    for {
		r := l % s
		if r == 0 {
			return s
		}
		s, l = r, s
	}
}

func main() {
    var i, j, g int
	fmt.Scanf("%d %d", &i, &j)
	if i < j {
        g = gcd(i, j)
	} else {
        g = gcd(j, i)
	}
	fmt.Println(i * (j/g))
}

File: week2_algorithmic_warmup/lcm.py
def gcd(smaller, bigger):
    while True:
        remainder = bigger % smaller
        if remainder == 0:
            return smaller
        bigger, smaller = smaller, remainder


if __name__ == '__main__':
    input_str = input()
    input_list = [int(i) for i in input_str.split(' ')]
    input_list.sort()
    num_gcd = gcd(input_list[0], input_list[1])
    if num_gcd == 1:
        print(input_list[0] * input_list[1])
    else:
        print(input_list[0] * (input_list[1]/num_gcd))
